zkir/Dialect/Field/Conversions/FieldToModArith/Extension/CubicExtensionField.cpp

```diff
@@ -0,0 +1,184 @@
+/* Copyright 2025 The ZKIR Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "zkir/Dialect/Field/Conversions/FieldToModArith/Extension/CubicExtensionField.h"
+
+#include "zkir/Dialect/ModArith/IR/ModArithOps.h"
+
+namespace mlir::zkir::field {
+
+CubicExtensionField::CubicExtensionField(ImplicitLocOpBuilder &b, Value xi)
+    : b_(b), xi_(xi) {}
+
+Value CubicExtensionField::mulByNonResidue(Value v) {
+  return b_.create<mod_arith::MulOp>(xi_, v);
+}
+
+SmallVector<Value, 3> CubicExtensionField::square(Value c0, Value c1,
+                                                  Value c2) {
+  // CH-SQR2 algorithm from "Multiplication and Squaring on Pairing-Friendly
+  // Fields" by Devegili, OhEigeartaigh, Scott, Dahab (Section 4).
+  // https://eprint.iacr.org/2006/471.pdf
+  //
+  // For a = c0 + c1*u + c2*u^2 where u^3 = xi:
+  //
+  // s0 = c0^2
```

Comment:
We use unicode(superscript) ² for better rendering! could you change something like below?

```suggestion
  // s0 = c0²
```